src/upgrade.py

```diff
@@ -301,3 +329,27 @@ def _check_server_upgradeability(self) -> None:
         instance = self.charm._get_unit_fqdn(f"{self.charm.app.name}/0")
         self.charm._mysql.verify_server_upgradable(instance=instance)
         logger.debug("MySQL server is upgradeable")
+
+    def _check_server_unsupported_downgrade(self) -> bool:
+        """Check error log for unsupported downgrade.
+
+        https://dev.mysql.com/doc/mysql-errors/8.0/en/server-error-reference.html
+        """
+        if log_content := self.charm._mysql.fetch_error_log():
+            return "MY-013171" in log_content
+
+        return False
+
+    def _reset_on_unsupported_downgrade(self, container: Container) -> None:
```

Comment:
thoughts on adding a simple check that this is not done if there is only 1 unit in the cluster?

Reply:
that's done on the calling method (L256)